src/components/PlaybackControls.tsx

```diff
@@ -0,0 +1,249 @@
+      return;
+    }
+    controller.gotoTime(time);
+  };
+
+  const playHandler = (timeOverride?: number) => {
+    const timeOverrideProvided = timeOverride !== undefined;
+    const newTime = timeOverrideProvided ? timeOverride : currentTime;
+    // this is the same as nextFrameUnavailable, but computing it here
+    // prevents a looping bug where playback gets stuck oscillating between
+    // first and last frame
+    const nextFrameIsPastLastFrame = newTime + timeStep >= lastFrameTime;
+    controller.pause();
+    if (nextFrameIsPastLastFrame) {
+      setPlaybackState({
+        isPlaying: false,
+        currentTime: firstFrameTime,
+      });
+      controller.gotoTime(firstFrameTime);
+    } else {
+      setPlaybackState({
+        isPlaying: true,
+        currentTime: newTime,
+      });
+      controller.playFromTime(newTime);
+      controller.resume();
+    }
+  };
+
+  const pauseHandler = () => {
+    controller.pause();
+    setPlaybackState({ ...playbackState, isPlaying: false });
+  };
+
+  const gotoNextFrame = () => {
+    controller.gotoTime(currentTime + timeStep);
+  };
+
+  const goToPreviousFrame = () => {
+    controller.gotoTime(currentTime - timeStep);
+  };
+
+  const handleTimeInputChange = (userInput: number | null): void => {
+    if (userInput !== null) {
+      setTimeInput(userInput as number);
+    }
+  };
+
+  const handleTimeInputKeyDown = (
+    event: React.KeyboardEvent<HTMLInputElement>
+  ): void => {
+    if (event.key === 'Enter' || event.key === 'Tab') {
+      if (typeof timeInput === 'number') {
+        setPlaybackState({
+          ...playbackState,
+          currentTime: timeInput / timeUnits.magnitude,
+        });
+        skipToTime(timeInput / timeUnits.magnitude);
+      }
+    }
+    if (event.key === 'Escape') {
+      const inputNumberComponent = event.target as HTMLElement;
+      inputNumberComponent.blur();
+    }
+  };
+
+  const handleSliderChange = (sliderValue: number | [number, number]): void => {
+    skipToTime(sliderValue as number);
+    if (isPlaying) {
+      controller.pause();
+    }
+  };
+
+  const handleSliderAfterChange = (value: number): void => {
+    if (isPlaying) {
+      playHandler(value);
+    } else {
+      setPlaybackState({ ...playbackState, currentTime: value });
+    }
+  };
+
+  return (
+    <div className="playback-controls">
+      <Tooltip
+        placement="top"
+        title={previousFrameUnavailable ? '' : 'Skip 1 frame back'}
+        color={TOOLTIP_COLOR}
+      >
+        <Button
+          id={'back-button'}
+          className={classNames([
+            'btn',
+            previousFrameUnavailable ? 'disabled' : '',
+          ])}
+          onClick={goToPreviousFrame}
+          disabled={previousFrameUnavailable}
+          icon={FrameBack}
+        ></Button>
+      </Tooltip>
+      <Tooltip
+        placement="top"
+        title={isPlaying ? 'Pause' : 'Play'}
+        color={TOOLTIP_COLOR}
```

Comment:
```suggestion
        color={TOOLTIP_COLOR}
        trigger=["hover", "focus"]
```
Recommend adding this to make tooltips respond to tab navigation too! (You'll want to update the other tooltips too though)